app/src/app/components/specmath-modal/modal.component.html

```diff
@@ -0,0 +1,105 @@
+<!-- Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this spec except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. -->
+
+<div class="modal-container">
+  <mat-dialog-content>
+    <div class="modal-step-header-container">
+      <div class="modal-step-header-labels">
+        <h3>
+          Merge specs <b class="step-indicator">{{ currentStep }}/4</b>
+        </h3>
+        <p class="mat-caption">
+          {{newSpecName}}
+        </p>
+      </div>
+      <button
+        id="modal-cancel-button"
+        mat-dialog-close
+        mat-icon-button
+        color="grey"
+        aria-label="Close modal button"
+      >
+        <mat-icon>close</mat-icon>
+      </button>
+    </div>
+    <mat-horizontal-stepper linear #stepper>
+      <mat-step>
+        <div class="modal-step-container">
+          <div class="modal-step-content">
+            <p class="mat-caption">
+              Enter the name of the new spec
+            </p>
+            <form class="spec-name-form">
+              <mat-form-field class="spec-name-field">
+                <input
+                  class="spec-name-input"
+                  type="text"
+                  matInput
+                  [formControl]="specNameFormControl"
+                  [(ngModel)]="newSpecName"
+                  placeholder="Spec name"
+                />
+                <span matSuffix>.yaml</span>
+                <mat-error *ngIf="specNameFormControl.hasError('required')">
+                  Naming your new spec spec is required
+                </mat-error>
+                <mat-error *ngIf="specNameFormControl.hasError('pattern')">
+                  The name can only contain alpha-numerical characters,
+                  underscores, and dashes
+                </mat-error>
+              </mat-form-field>
+            </form>
+          </div>
+          <div class="modal-step-controls-container">
+            <button
+              id="modal-step-1-back"
+              mat-button
+              matStepperPrevious
+              class="modal-step-button-secondary"
+              aria-label="Back button for step 1"
+            >
+              Back
```

Comment:
if this is the first step, should the back button say something like "close" or "cancel"?

Reply:
done 👍